lnd.go

```diff
@@ -1386,10 +1398,28 @@ func waitForWalletPassword(cfg *Config,
 		netDir := btcwallet.NetworkDir(
 			chainConfig.ChainDir, cfg.ActiveNetParams.Params,
 		)
-		loader := wallet.NewLoader(
-			cfg.ActiveNetParams.Params, netDir, !cfg.SyncFreelist,
-			cfg.DB.Bolt.DBTimeout, recoveryWindow,
+
+		var (
+			loader *lnwallet.Loader
+			err    error
 		)
+		if cfg.Cluster.EnableLeaderElection {
+			loader, err = lnwallet.NewLoader(
+				cfg.ActiveNetParams.Params, recoveryWindow,
+				lnwallet.LoaderWithExternalDB(remoteDB),
```

Comment:
nit: only create the `LoaderOptionFunc` inside the if/else, then call the `lnwallet.NewLoader` afterwards?

Reply:
Indeed, done.